File: msgraph-mail/typescript/src/users/inferenceClassification/overridesResponse.ts
import {SerializationWriter, ParseNode, Parsable} from '@microsoft/kiota-abstractions';
import {InferenceClassificationOverride} from './inferenceClassificationOverride';

export class OverridesResponse implements Parsable<OverridesResponse> {
    public value?: InferenceClassificationOverride[] | undefined;
    public nextLink?: string | undefined;
    public serialize (writer: SerializationWriter) : void {
        writer.writeCollectionOfObjectValues<InferenceClassificationOverride>("value", this.value);
        writer.writeStringValue("nextLink", this.nextLink);
    };
    public deserializeFields () : Map<string, (item: OverridesResponse, node: ParseNode) => void> {
        return new Map<string, (item: OverridesResponse, node: ParseNode) => void>([
            ["value", (o, n) => { o.value = n.getCollectionOfObjectValues<InferenceClassificationOverride>(InferenceClassificationOverride); }],
            ["nextLink", (o, n) => { o.nextLink = n.getStringValue(); }],
        ]);
    };
}

File: msgraph-mail/dotnet/Users/InferenceClassification/OverridesResponse.cs
using Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.Linq;
namespace Graphdotnetv4.Users.InferenceClassification {
    public class OverridesResponse : IParsable<OverridesResponse> {
        public List<InferenceClassificationOverride> Value { get; set; }
        public string NextLink { get; set; }
        public IDictionary<string, Action<OverridesResponse, IParseNode>> DeserializeFields => new Dictionary<string, Action<OverridesResponse, IParseNode>> {
            {
                "value", (o,n) => { o.Value = n.GetCollectionOfObjectValues<InferenceClassificationOverride>().ToList(); }
            },
            {
                "nextLink", (o,n) => { o.NextLink = n.GetStringValue(); }
            },
        };
        public void Serialize(ISerializationWriter writer) {
            writer.WriteCollectionOfObjectValues<InferenceClassificationOverride>("value", Value);
            writer.WriteStringValue("nextLink", NextLink);
        }
    }
}
